ideascube/mediacenter/views.py

```diff
@@ -16,6 +16,9 @@
 from ideascube.decorators import staff_member_required
 from ideascube.mixins import FilterableViewMixin, OrderableViewMixin
 
+# Do not import directly Catalog as we want to mock it for test.
```

Comment:
> Tests should no be dependent from where the class is imported. They should mock the class only where it is defined (ideascube.serveradmin.catalog)

Nope: https://docs.python.org/3/library/unittest.mock.html#where-to-patch

> The basic principle is that you patch where an object is looked up, which is not necessarily the same place as where it is defined.

It is perfectly acceptable for a test of `B` to mock things only in `B` and not touch at `A` where the mocked thing is defined.

And what you did is fine too.

When I started these threads I had just missed a simple fact: you're mocking in a fixture, not in the individual tests, so you have to do it this way.